test/gurka/test_expansion.cc

```diff
@@ -0,0 +1,132 @@
+
+TEST_F(ExpansionTest, Isochrone) {
+  // test Dijkstra expansion
+  const auto& center_node = expansion_map.nodes["A"];
+  // remember there's no way to get less than 10 km or mins expansion (buffer in isochrones)
+  const std::string req = R"({"locations":[{"lat":)" + std::to_string(center_node.lat()) +
+                          R"(,"lon":)" + std::to_string(center_node.lng()) +
+                          R"(}],"costing":"auto","contours":[{"distance":0.05}]})";
+  std::string res;
+  auto api = gurka::do_action(Options::expansion, expansion_map, req, nullptr, &res);
+
+  rapidjson::Document res_doc;
+  res_doc.Parse(res.c_str());
+
+  // 11 because there's a one-way
+  check_props(res_doc, 11, {"distances", "durations", "costs"});
+}
+
+TEST_F(ExpansionTest, IsochroneNoOpposites) {
+  // test Dijkstra expansion and skip collecting more expensive opposite edges
+  const auto& center_node = expansion_map.nodes["A"];
+  // remember there's no way to get less than 10 km or mins expansion (buffer in isochrones)
+  const std::string req = R"({"skip_opposites":true,"locations":[{"lat":)" +
+                          std::to_string(center_node.lat()) + R"(,"lon":)" +
+                          std::to_string(center_node.lng()) +
+                          R"(}],"costing":"auto","contours":[{"distance":0.05}]})";
+  std::string res;
+  auto api = gurka::do_action(Options::expansion, expansion_map, req, nullptr, &res);
+
+  rapidjson::Document res_doc;
+  res_doc.Parse(res.c_str());
+
+  check_props(res_doc, 6, {"distances", "durations", "costs"});
+}
+
+TEST_F(ExpansionTest, Routing) {
+  // test AStar expansion
+  const std::unordered_map<std::string, std::string> options = {{"/action", "route"}};
+  std::vector<midgard::PointLL> lls;
+  for (const auto& node_name : {"E", "H"}) {
+    lls.push_back(expansion_map.nodes.at(node_name));
+  }
+  const auto req =
+      gurka::detail::build_valhalla_request(std::string("locations"), lls, "auto", options);
+  std::string res;
+  auto api = gurka::do_action(Options::expansion, expansion_map, req, nullptr, &res);
+
+  rapidjson::Document res_doc;
+  res_doc.Parse(res.c_str());
+
+  check_props(res_doc, 23, {"statuses", "edge_ids"});
+}
+
+TEST_F(ExpansionTest, RoutingNoOpposites) {
+  // test AStar expansion
+  const std::unordered_map<std::string, std::string> options = {{"/action", "route"},
+                                                                {"/skip_opposites", "true"}};
+  std::vector<midgard::PointLL> lls;
+  for (const auto& node_name : {"E", "H"}) {
+    lls.push_back(expansion_map.nodes.at(node_name));
+  }
+  const auto req =
+      gurka::detail::build_valhalla_request(std::string("locations"), lls, "auto", options);
+  std::string res;
+  auto api = gurka::do_action(Options::expansion, expansion_map, req, nullptr, &res);
+
+  rapidjson::Document res_doc;
+  res_doc.Parse(res.c_str());
+
+  check_props(res_doc, 23, {"statuses", "edge_ids"});
```

Comment:
it's expected that bidir a* branches don't circle back on the opposites right (i.e. both `skip_opposites` true & false give 23 directed edges)? not entirely sure about that..

Reply:
must've been smth off, now I get 16 edges instead of 23, sounds more reasonable